lib/features/mailbox/data/repository/mailbox_repository_impl.dart

```diff
@@ -41,59 +44,100 @@ class MailboxRepositoryImpl extends MailboxRepository {
-        ]);
-      }
-    } else {
-      final mailboxResponse = await mapDataSource[DataSourceType.network]!.getAllMailbox(session, accountId);
+    final jmapMailboxResponse = await _getAllMailboxFromJMAP(accountId, session);
+    if (jmapMailboxResponse == null) {
+      return;
+    }
 
-      await Future.wait([
-        mapDataSource[DataSourceType.local]!.update(accountId, session.username, created: mailboxResponse.mailboxes),
-        if (mailboxResponse.state != null)
-          stateDataSource.saveState(accountId, session.username, mailboxResponse.state!.toStateCache(StateType.mailbox)),
-      ]);
+    await _syncNewInCache(
+      accountId,
+      session.username,
+      jmapMailboxResponse.mailboxes,
+      jmapMailboxResponse.state!);
+
+    yield jmapMailboxResponse;
+  }
+
+  Future<CacheMailboxResponse?> _getAllMailboxFromCache(AccountId accountId, UserName userName) async {
+    try {
+      final cacheMailboxResponse = await Future.wait([
+        mapDataSource[DataSourceType.local]!.getAllMailboxCache(accountId, userName),
+        stateDataSource.getState(accountId, userName, StateType.mailbox)
+      ], eagerError: true).then((List response) => CacheMailboxResponse(mailboxes: response.first, state: response.last));
+      log('MailboxRepositoryImpl::_getAllMailboxFromCache: MAILBOX_CACHED = ${cacheMailboxResponse.mailboxes.length} | STATE_CACHED = ${cacheMailboxResponse.state?.value}');
+      return cacheMailboxResponse;
+    } catch (e) {
+      logError('MailboxRepositoryImpl::_getAllMailboxFromCache: Exception: $e');
+      return null;
+    }
+  }
+
+  Future<JmapMailboxResponse?> _getAllMailboxFromJMAP(
+    AccountId accountId,
+    Session session,
+    {Properties? properties}
+  ) async {
+    try {
+      final jmapMailboxResponse = await mapDataSource[DataSourceType.network]!.getAllMailbox(
+        session,
+        accountId,
+        properties: properties);
+      log('MailboxRepositoryImpl::_getAllMailboxFromJMAP: MAILBOX_NETWORK = ${jmapMailboxResponse.mailboxes.length} | STATE_NETWORK = ${jmapMailboxResponse.state}');
+      return jmapMailboxResponse;
+    } catch (e) {
+      logError('MailboxRepositoryImpl::_getAllMailboxFromJMAP: Exception: $e');
+      return null;
     }
+  }
 
-    final newMailboxResponse = await Future.wait([
-      mapDataSource[DataSourceType.local]!.getAllMailboxCache(accountId, session.username),
-      stateDataSource.getState(accountId, session.username, StateType.mailbox)
-    ]).then((List response) {
-      return MailboxResponse(mailboxes: response.first, state: response.last);
-    });
+  Future<void> _syncNewInCache(
+    AccountId accountId,
+    UserName userName,
+    List<Mailbox> newMailboxes,
+    State newState,
+  ) async {
+    await Future.wait([
+      _syncNewMailboxesInCache(accountId, userName, newMailboxes),
+      _syncNewMailboxStateInCache(accountId, userName, newState)
```

Comment:
please await in `syncMailbox` before `synsMailboxState`

Reply:
oke